turbinia/workers/analysis/windows_acct.py

```diff
@@ -0,0 +1,139 @@
+# -*- coding: utf-8 -*-
+# Copyright 2021 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Task for analysing Windows account passwords."""
+
+import os
+import subprocess
+
+from turbinia import TurbiniaException
+from turbinia.evidence import EvidenceState as state
+from turbinia.evidence import ReportText
+from turbinia.lib import text_formatter as fmt
+from turbinia.lib.utils import bruteforce_password_hashes
+from turbinia.lib.utils import extract_artifacts
+from turbinia.workers import Priority
+from turbinia.workers import TurbiniaTask
+
+
+class WindowsAccountAnalysisTask(TurbiniaTask):
+  """Task to analyze a Linux password file."""
+
+  REQUIRED_STATES = [state.ATTACHED, state.DECOMPRESSED]
+
+  def run(self, evidence, result):
+    """Run the Windows Account worker.
+
+    Args:
+        evidence (Evidence object):  The evidence to process
+        result (TurbiniaTaskResult): The object to place task results into.
+    Returns:
+        TurbiniaTaskResult object.
+    """
+    # Where to store the resulting output file.
+    output_file_name = 'windows_account_analysis.txt'
+    output_file_path = os.path.join(self.output_dir, output_file_name)
+
+    # What type of evidence we should output.
+    output_evidence = ReportText(source_path=output_file_path)
+
+    try:
+      location = self._collect_windows_files(evidence)
+    except TurbiniaException as e:
+      result.close(
+          self, success=True,
+          status='No Windows account files found: {0:s}'.format(str(e)))
+      return result
+    (creds, hashnames) = self._extract_windows_hashes(
+        os.path.join(location, 'Windows', 'System32', 'config'))
+    (report, priority, summary) = self._analyse_windows_creds(creds, hashnames)
+    output_evidence.text_data = report
+    result.report_priority = priority
+    result.report_data = report
+
+    # Write the report to the output file.
+    with open(output_file_path, 'wb') as fh:
+      fh.write(output_evidence.text_data.encode('utf-8'))
+
+    # Add the resulting evidence to the result object.
+    result.add_evidence(output_evidence, evidence.config)
+    result.close(self, success=True, status=summary)
+    return result
+
+  def _collect_windows_files(self, evidence):
+    try:
```

Comment:
I wonder if we should add this try/except block to inline in the `extract_artifacts()` method so that the other callers also get the advantage of this?

Reply:
Actually it is! This just captures the already raised exception and re-raises it. 
Can probably be removed but I found it useful to have the specific exception string while debugging